### DISCUSSION TITLE
Authentication Remember Me

### DISCUSSION THREAD

**Discussion by scottkuhl** [Azure/static-web-apps#770] [Q&A] [answered]
Is there a way to keep the user signed in across sessions similar to a Remember Me checkbox?

When I come back to the app it usually asks me to sign in again.

FYI, using Blazor Web Assembly.

**Comment by scottkuhl** [comment]
It does not seem like the team really monitors any of these discussions.  I am seeing very little engagement here.

I did find an open issue here: #213

**Comment by scottkuhl** [comment, marked as answer]
I will answer my own question for future reference for others.

The authentication is stored in a cookie that persists across browser sessions.

<img width="231" alt="image" src="https://user-images.githubusercontent.com/4465616/189498854-a75ea8cd-af3e-44bc-8c5a-5f7972867ea3.png">

You can access JSON data about the logged in user at /.auth/me:

```
{
  "clientPrincipal": {
    "identityProvider": "aad",
    "userId": "xxxxxxxxxxxxxxx",
    "userDetails": "user@domain.com",
    "userRoles": [
      "anonymous",
      "authenticated"
    ]
  }
}
```

How and when you use that information will vary based on your front end.  This article uses it for Blazor WebAssembly:

[Authenticate Blazor WebAssembly with Azure Static Web Apps](https://anthonychu.ca/post/blazor-auth-azure-static-web-apps/)

Here is a link to the code that reads that data and authenticates the user:

https://github.com/anthonychu/blazor-auth-static-web-apps/blob/main/src/StaticWebAppsAuthenticationExtensions/StaticWebAppsAuthenticationStateProvider.cs

**Reply by zola-25** [reply]
Thank you I might be needing this
